src/megatron/energon/flavors/webdataset/indexing.py

```diff
@@ -140,39 +178,69 @@ def append_part(
             (tar_file_id, sample_index, part_name, content_byte_offset, content_byte_size),
         )
 
+    def append_media_metadata(
```

Comment:
Should we also have an add batch method (e.g. collecting 1024 entries, then adding all at once)? Is this efficient enough?